In [73]:
#!pip install gdown
import gdown
from scipy import stats

In [74]:
file_id = '1yB5qSBOLl96Y563nIewKOU8RN_gsY3dO'  # Make sure it's a string
gdown.download(f'https://drive.google.com/uc?id={file_id}', 'data.csv', quiet=False)

Downloading...
From: https://drive.google.com/uc?id=1yB5qSBOLl96Y563nIewKOU8RN_gsY3dO
To: c:\Users\ncc777\Desktop\Data analysis\a_env\file_3\data.csv
100%|██████████| 52.0k/52.0k [00:00<00:00, 1.40MB/s]


'data.csv'

In [75]:
import pandas as pd
import numpy as np


In [76]:
# Read the downloaded CSV file
df = pd.read_csv('data.csv')
df.head()

Unnamed: 0    country food_category  consumption  co2_emission
0           1  Argentina          pork        10.51         37.20
1           2  Argentina       poultry        38.66         41.53
2           3  Argentina          beef        55.48       1712.00
3           4  Argentina     lamb_goat         1.56         54.63
4           5  Argentina          fish         4.36          6.96

In [77]:
# Select Argentina beef consumption
arg_beef = df[(df["country"]=="Argentina") & (df["food_category"]=="beef")]
arg_beef

Unnamed: 0    country food_category  consumption  co2_emission
2           3  Argentina          beef        55.48        1712.0

In [78]:
arg_consump = arg_beef["consumption"]
arg_consump

2    55.48
Name: consumption, dtype: float64

In [79]:
arg_consump_mean = arg_consump.mean()
print(arg_consump_mean)
print("Mean Beef Consumption (Argentina):", arg_consump_mean)

55.48
Mean Beef Consumption (Argentina): 55.48


In [80]:
df.country.unique()

array(['Argentina', 'Australia', 'Albania', 'Iceland', 'New Zealand',
       'USA', 'Uruguay', 'Luxembourg', 'Brazil', 'Kazakhstan', 'Sweden',
       'Bermuda', 'Denmark', 'Finland', 'Ireland', 'Greece', 'France',
       'Canada', 'Norway', 'Hong Kong SAR. China', 'French Polynesia',
       'Israel', 'Switzerland', 'Netherlands', 'Kuwait', 'United Kingdom',
       'Austria', 'Oman', 'Italy', 'Bahamas', 'Portugal', 'Malta',
       'Armenia', 'Slovenia', 'Chile', 'Venezuela', 'Belgium', 'Germany',
       'Russia', 'Croatia', 'Belarus', 'Spain', 'Paraguay',
       'New Caledonia', 'South Africa', 'Barbados', 'Lithuania', 'Turkey',
       'Estonia', 'Mexico', 'Costa Rica', 'Bolivia', 'Ecuador', 'Panama',
       'Czech Republic', 'Romania', 'Colombia', 'Maldives', 'Cyprus',
       'Serbia', 'United Arab Emirates', 'Algeria', 'Ukraine', 'Pakistan',
       'Swaziland', 'Latvia', 'Bosnia and Herzegovina', 'Fiji',
       'South Korea', 'Poland', 'Saudi Arabia', 'Botswana', 'Macedonia',
       '

In [81]:
# check unique value for food category
df.food_category.unique()

array(['pork', 'poultry', 'beef', 'lamb_goat', 'fish', 'eggs', 'dairy',
       'wheat', 'rice', 'soybeans', 'nuts'], dtype=object)

In [82]:
# Select Bangladesh beef consumption
bang_beef = df[(df['country'] == 'Bangladesh') & (df['food_category'] == 'beef')]
bang_beef 

Unnamed: 0     country food_category  consumption  co2_emission
1421        1422  Bangladesh          beef         1.28          39.5

In [83]:

bang_consump = bang_beef["consumption"]
bang_consump

1421    1.28
Name: consumption, dtype: float64

In [84]:
bang_consump_mean = bang_consump.mean()
print(bang_consump_mean)
print("Mean Beef Consumption ('Bangladesh'):", bang_consump_mean)

1.28
Mean Beef Consumption ('Bangladesh'): 1.28


Since the Mean beef consumption in Argentina(55.48) ≠ Mean beef consumption in Bangladesh(1.28), then we reject the null hypothesis and conclude that the beef consumption in Argentina are different from that in Bangladesh

In [94]:
arg_beef_samples = np.random.normal(loc=arg_consump_mean, scale=5, size=30)
arg_beef_samples

array([52.37185713, 58.60131181, 57.55029944, 54.98708963, 59.36217535,
       61.08290353, 49.11417456, 49.99066446, 57.2220516 , 52.80659571,
       57.11178604, 54.74687329, 49.40858539, 62.9818672 , 53.15486785,
       56.79399209, 57.28123192, 60.46359881, 62.72055991, 57.36457537,
       48.23135415, 59.16321912, 64.38248604, 52.60483088, 53.33652356,
       50.42719529, 50.96160677, 50.96607095, 63.34697535, 59.6872832 ])

In [95]:
bang_beef_samples = np.random.normal(loc=bang_consump_mean, scale=5, size=30)
bang_beef_samples

array([  2.92174532,   0.519579  ,   0.54580128,  -3.99076643,
        -1.03042027,  -3.71554602,  11.38856611,  -3.57568644,
         8.67101704,   3.75848994,  -4.70506812,  -4.4593978 ,
        -3.25220776,  -1.57908205,   7.17403124,  -3.8252908 ,
        10.47925658,  -4.98251917,   2.27556637,   0.12348366,
        10.84464595,   2.87037382,   5.24161003,  -3.30838863,
         0.12323172,   4.9243826 , -13.04966733,   7.95011236,
        -0.81305697,  -1.78201721])

In [96]:
# Two-sample t-test
t_stat, p_val = stats.ttest_ind(arg_beef_samples, bang_beef_samples)


In [ ]:
# one-sample t-test
#t_stat, p_val = stats.ttest_1samp(arg_beef_samples, bang_beef_samples)

In [97]:
print("T-statistic:", t_stat)

T-statistic: 40.96439423292167


Interpretation

* Since the P-value is less than 0.05, we reject null hypothesis, So we say Mean beef consumption in Argentina(55.48) is significantly different from Mean beef consumption in Bangladesh(1.28).